content/docs/rfcs/44/README.md

```diff
@@ -0,0 +1,272 @@
+
+# Specification
+
+Nodes supporting 44/WAKU2-DANDELION MUST either be in stem state or in fluff state.
+This does not include relaying messages originated in $v$, for which $v$ SHOULD always be in stem state.
+
+## Choosing the State
+
+On startup and when a new epoch starts,
+node $v$ randomly selects a number $r$ between 0 and 1.
+If $r < q$, for $q = 0.2$, the node enters fluff state, otherwise, it enters stem state.
+
+New epochs start when `unixtime` (in seconds) $\equiv 0 \mod 600$,
+corresponding to 10 minute epochs.
+
+## Stem State
+
+On entering stem state,
+nodes supporting 44/WAKU2-DANDELION MUST randomly select two nodes for each pubsub topic from the respective gossipsub mesh node set.
+These nodes are referred to as stem relays.
+Stem relays MUST support [19/WAKU2-LIGHTPUSH](/spec/19/).
+If a chosen peer does not support [19/WAKU2-LIGHTPUSH](/spec/19/),
+the node SHOULD switch to fluff state.
+(We may update this strategy in future versions of this document.)
+
+Further, the node establishes a map that maps each incoming stem connection
+to one of its stem relays chosen at random (but fixed per epoch).
+Incoming stem connections are identified by the [Peer IDs](https://docs.libp2p.io/concepts/peers/#peer-id/)
+of peers the node receives [19/WAKU2-LIGHTPUSH](/spec/19/) messages from.
+Incoming [19/WAKU2-LIGHTPUSH](/spec/19/) connections from peers that do not support 44/WAKU2-DANDELION are treated in the same way.
+This makes the protocol simpler, increases the anonymity set, and offers Dandelion anonymity properties to such peers, too.
+
+The node itself is mapped in the same way, so that all messages originated by the node are relayed via a per-epoch-fixed Dandelion relay, too.
+
+While in stem state, nodes MUST relay stem messages to the respective stem relay.
+Received fluff messages MUST be relayed as specified in the fluff state section.
+
+The stem protocol ([19/WAKU2-LIGHTPUSH](/spec/19/)) is independent of the fluff protocol ([Waku Relay](/spec/11/)).
+While in stem state, nodes MUST NOT gossip about stem messages,
+and MUST NOT send control messages related to stem messages.
+(An existing gossipsub implementation does *not* have to be adjusted to not send gossip about stem messages,
+ because these messages are only handed to gossipsub once they enter fluff phase.)
+
+### Fail Safe
+
+Nodes $v$ in stem state SHOULD store messages attached with a random timer between $t_1 = 5 * 100ms$ and $t_2 = 2 * t_1$.
+This time interval is chosen because
+
+* we assume $100\,ms$ as an average per hop delay, and
+* using $q=0.2$ will lead to an expected number of 5 stem hops per message.
+
+If $v$ does not receive a given message via Waku Relay (fluff) before the respective timer runs out,
+$v$ will disseminate the message via Waku Relay.
+
+## Fluff State
+
+In fluff state, nodes operate as usual Waku Relay nodes,
+with the exception of adding a random delay between $0$ and $100\,ms$ before relaying a message.
```

Comment:
One of the scores in the peer scoring is related to the fast delivery of the messages i.e., `p2` `First Message Deliveries `, by introducing this delay, peers may lose the chance to get that score.